lib/cose/messages/sign1.ex

```diff
@@ -19,6 +19,28 @@ defmodule COSE.Messages.Sign1 do
     CBOR.encode(%CBOR.Tag{tag: 18, value: value})
   end
 
+  def sign_encode(:es256, msg, key) do
+    msg = sign(:es256, msg, key, <<>>)
+
+    value = [
+      COSE.Headers.tag_phdr(msg.phdr),
+      msg.uhdr,
+      msg.payload,
+      msg.signature
+    ]
+
+    CBOR.encode(%CBOR.Tag{tag: 18, value: value})
+  end
+
+  def sign(:es256, msg, private_key, external_aad) do
+    to_be_signed = CBOR.encode(sig_structure(msg, external_aad))
+
+    %__MODULE__{
+      msg
+      | signature: COSE.Keys.ECDSA.sign(:es256, to_be_signed, private_key)
+    }
+  end
+
```

Comment:
Some of the introduced functions duplicate a fair amount of the body, including `sign_encode(:es256 ...)`, `sign(:es256, ...)`, etc. Could it be improved?